codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -1065,6 +1069,7 @@ public void remove() {
     private final class ValueIterator implements Iterator<V> {
         private final K name;
         private final int hash;
+        private HeaderEntry<K, V> removalPrevious;
```

Comment:
@normanmaurer - an alternative is to not track the "previous previous" entry, and just iterate through the bucket when it is time to remove. If did this we could also get rid of "previous" because we would have to iterate through the bucket each time anyways. I added extra state for now to trade some memory to avoid duplicate iteration on removal.